### ISSUE TITLE
Nuget path-sources: Couldn't get package details...

### ISSUE THREAD

**Issue by devboy** [fsprojects/Paket#824]
Given the directory:

```
/Users/devboy/some-lib/bin
└── Some.Lib.12.2.0.nupkg
```

the `paket.dependencies` file:

```
source ~/some-lib/bin

nuget Some.Lib
```

running the command `mono .paket/paket.exe update` produces the following:

```
Paket version 1.4.4.0
Resolving packages:
 - Some.Lib 12.2.0
Paket failed with:
	Couldn't get package details for package Some.Lib on any of ["~/some-lib/bin"].
```

It detects that the version 12.2.0 exists, but fails for some reason.

**Comment by devboy** [comment]
it works on the clr, but on mono I get the following stacktrace when running with `--verbose`:
```
StackTrace:
    at Paket.NuGetV2+tryNext@592-1.Invoke (System.String message) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.PrintfImpl+StringPrintfEnv`1[System.Tuple`2[Paket.PackageSources+PackageSource,Paket.NuGetV2+NugetPackageCache]].Finalize () [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.PrintfImpl+Final2@235[Microsoft.FSharp.Core.Unit,System.String,System.Tuple`2[Paket.PackageSources+PackageSource,Paket.NuGetV2+NugetPackageCache],System.String,Microsoft.FSharp.Collections.FSharpList`1[System.String]].Invoke (Microsoft.FSharp.Core.FSharpFunc`2 env, System.String a, Microsoft.FSharp.Collections.FSharpList`1 b) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.OptimizedClosures+Invoke@3041-1[System.String,Microsoft.FSharp.Collections.FSharpList`1[System.String],System.Tuple`2[Paket.PackageSources+PackageSource,Paket.NuGetV2+NugetPackageCache],Microsoft.FSharp.Core.FSharpFunc`2[Microsoft.FSharp.Core.Unit,Microsoft.FSharp.Core.PrintfImpl+PrintfEnv`3[Microsoft.FSharp.Core.Unit,System.String,System.Tuple`2[Paket.PackageSources+PackageSource,Paket.NuGetV2+NugetPackageCache]]]].Invoke (System.String u, Microsoft.FSharp.Collections.FSharpList`1 v) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.OptimizedClosures+Invoke@3028[Microsoft.FSharp.Collections.FSharpList`1[System.String],System.Tuple`2[Paket.PackageSources+PackageSource,Paket.NuGetV2+NugetPackageCache],System.String].Invoke (Microsoft.FSharp.Collections.FSharpList`1 u) [0x00000] in <filename unknown>:0
  at Paket.NuGetV2+tryNext@592-3.Invoke (Microsoft.FSharp.Collections.FSharpList`1 arg20) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.FSharpFunc`2[System.String,Microsoft.FSharp.Collections.FSharpList`1[System.String]].InvokeFast[Tuple`2] (Microsoft.FSharp.Core.FSharpFunc`2 func, System.String arg1, Microsoft.FSharp.Collections.FSharpList`1 arg2) [0x00000] in <filename unknown>:0
  at Paket.NuGetV2.tryNext@571 (System.String root, Boolean force, Microsoft.FSharp.Collections.FSharpList`1 sources, Paket.SemVerInfo version, System.String activePatternResult, Microsoft.FSharp.Collections.FSharpList`1 xs) [0x00000] in <filename unknown>:0
  at Paket.NuGetV2.tryNext@571 (System.String root, Boolean force, Microsoft.FSharp.Collections.FSharpList`1 sources, Paket.SemVerInfo version, System.String activePatternResult, Microsoft.FSharp.Collections.FSharpList`1 xs) [0x00000] in <filename unknown>:0
  at Paket.NuGetV2.tryNext@571 (System.String root, Boolean force, Microsoft.FSharp.Collections.FSharpList`1 sources, Paket.SemVerInfo version, System.String activePatternResult, Microsoft.FSharp.Collections.FSharpList`1 xs) [0x00000] in <filename unknown>:0
  at Paket.NuGetV2.GetPackageDetails (System.String root, Boolean force, Microsoft.FSharp.Collections.FSharpList`1 sources, Paket.PackageName _arg1, Paket.SemVerInfo version) [0x00000] in <filename unknown>:0
  at <StartupCode$Paket-Core>.$DependenciesFile+Resolve@313-3.Invoke (Microsoft.FSharp.Collections.FSharpList`1 sources, Paket.PackageName arg30@, Paket.SemVerInfo version) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.FSharpFunc`2[Microsoft.FSharp.Collections.FSharpList`1[Paket.PackageSources+PackageSource],Paket.Domain+PackageName].InvokeFast[SemVerInfo,PackageDetails] (Microsoft.FSharp.Core.FSharpFunc`2 func, Microsoft.FSharp.Collections.FSharpList`1 arg1, Paket.PackageName arg2, Paket.SemVerInfo arg3) [0x00000] in <filename unknown>:0
  at Paket.PackageResolver.getExploredPackage@155 (Microsoft.FSharp.Collections.FSharpList`1 globalFrameworkRestrictions, Microsoft.FSharp.Core.FSharpFunc`2 getPackageDetailsF, System.Collections.Generic.Dictionary`2 exploredPackages, Paket.PackageRequirement dependency, Paket.SemVerInfo version) [0x00000] in <filename unknown>:0
  at Paket.PackageResolver+tryToImprove@298-2.Invoke (Paket.SemVerInfo versionToExplore) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.OptimizedClosures+Adapt@3034[System.Tuple`2[System.Boolean,Paket.PackageResolver+Resolution],Paket.SemVerInfo,System.Tuple`2[System.Boolean,Paket.PackageResolver+Resolution]].Invoke (System.Tuple`2 , Paket.SemVerInfo ) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Collections.ListModule.loop@142-20[SemVerInfo,Tuple`2] (Microsoft.FSharp.Core.FSharpFunc`3 f, System.Tuple`2 s, Microsoft.FSharp.Collections.FSharpList`1 xs) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Collections.ListModule.Fold[SemVerInfo,Tuple`2] (Microsoft.FSharp.Core.FSharpFunc`2 folder, System.Tuple`2 state, Microsoft.FSharp.Collections.FSharpList`1 list) [0x00000] in <filename unknown>:0
  at Paket.PackageResolver.tryToImprove@296 (Microsoft.FSharp.Collections.FSharpList`1 globalFrameworkRestrictions, Microsoft.FSharp.Core.FSharpFunc`2 getVersionsF, Microsoft.FSharp.Core.FSharpFunc`2 getPackageDetailsF, System.Collections.Generic.Dictionary`2 exploredPackages, System.Collections.Generic.Dictionary`2 allVersions, System.Collections.Generic.Dictionary`2 conflictHistory, Microsoft.FSharp.Collections.FSharpList`1 selectedPackageVersions, Microsoft.FSharp.Collections.FSharpSet`1 openRequirements, Microsoft.FSharp.Collections.FSharpMap`2 filteredVersions, Microsoft.FSharp.Collections.FSharpSet`1 closedRequirements, Paket.PackageRequirement currentRequirement, Microsoft.FSharp.Core.FSharpRef`1 globalOverride, Microsoft.FSharp.Collections.FSharpList`1 sortedVersions, Boolean useUnlisted) [0x00000] in <filename unknown>:0
  at Paket.PackageResolver.step@199 (Microsoft.FSharp.Collections.FSharpList`1 globalFrameworkRestrictions, Microsoft.FSharp.Core.FSharpFunc`2 getVersionsF, Microsoft.FSharp.Core.FSharpFunc`2 getPackageDetailsF, System.Collections.Generic.Dictionary`2 exploredPackages, System.Collections.Generic.Dictionary`2 allVersions, System.Collections.Generic.Dictionary`2 conflictHistory, Microsoft.FSharp.Collections.FSharpMap`2 filteredVersions, Microsoft.FSharp.Collections.FSharpList`1 selectedPackageVersions, Microsoft.FSharp.Collections.FSharpSet`1 closedRequirements, Microsoft.FSharp.Collections.FSharpSet`1 openRequirements) [0x00000] in <filename unknown>:0
  at Paket.PackageResolver.Resolve (Microsoft.FSharp.Core.FSharpFunc`2 getVersionsF, Microsoft.FSharp.Core.FSharpFunc`2 getPackageDetailsF, Microsoft.FSharp.Collections.FSharpList`1 globalFrameworkRestrictions, Microsoft.FSharp.Collections.FSharpList`1 rootDependencies) [0x00000] in <filename unknown>:0
  at Paket.DependenciesFile.Resolve (Microsoft.FSharp.Core.FSharpFunc`2 getSha1, Microsoft.FSharp.Core.FSharpFunc`2 getVersionF, Microsoft.FSharp.Core.FSharpFunc`2 getPackageDetailsF) [0x00000] in <filename unknown>:0
  at Paket.DependenciesFile.Resolve (Boolean force) [0x00000] in <filename unknown>:0
  at Paket.UpdateProcess.SelectiveUpdate (Paket.DependenciesFile dependenciesFile, Microsoft.FSharp.Core.FSharpOption`1 exclude, Boolean force) [0x00000] in <filename unknown>:0
  at Paket.UpdateProcess.SmartInstall (System.String dependenciesFileName, Microsoft.FSharp.Core.FSharpOption`1 exclude, Boolean force, Boolean hard, Boolean withBindingRedirects) [0x00000] in <filename unknown>:0
  at Paket.UpdateProcess.Update (System.String dependenciesFileName, Boolean force, Boolean hard, Boolean withBindingRedirects) [0x00000] in <filename unknown>:0
  at <StartupCode$Paket-Core>.$PublicAPI+Update@154.Invoke (Microsoft.FSharp.Core.Unit unitVar0) [0x00000] in <filename unknown>:0
  at Paket.Utils.RunInLockedAccessMode[Unit] (System.String rootFolder, Microsoft.FSharp.Core.FSharpFunc`2 action) [0x00000] in <filename unknown>:0
```

**Comment by forki** [comment]
I could not reproduce but hope that https://github.com/fsprojects/Paket/commit/15704b41d8c19dcdf46e2f32071c3a75f7b180d2 fixes it.

**Comment by devboy** [comment]
Now I am getting the following exception:

```
Source './nuget' exception: System.ArgumentOutOfRangeException: Argument is out of range.
  at System.DateTime.op_Subtraction (DateTime d, TimeSpan t) [0x00000] in <filename unknown>:0
  at System.DateTimeOffset.get_UtcDateTime () [0x00000] in <filename unknown>:0
  at System.DateTimeOffset..ctor (DateTime dateTime) [0x00000] in <filename unknown>:0
  at System.DateTimeOffset.op_Implicit (DateTime dateTime) [0x00000] in <filename unknown>:0
  at System.IO.Compression.ZipArchiveEntry.get_LastWriteTime () [0x00000] in <filename unknown>:0
  at System.IO.Compression.ZipFileExtensions.ExtractToFile (System.IO.Compression.ZipArchiveEntry source, System.String destinationFileName, Boolean overwrite) [0x00000] in <filename unknown>:0
  at Paket.NuGetV2+getDetailsFromLocalFile@337-2.Invoke (System.IO.Compression.ZipArchive _arg2) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Control.AsyncBuilderImpl+callA@803[Paket.NuGetV2+NugetPackageCache,System.IO.Compression.ZipArchive].Invoke (Microsoft.FSharp.Control.AsyncParams`1 args) [0x00000] in <filename unknown>:0
Trying source 'https://wooga.artifactoryonline.com/wooga/api/nuget/nuget-private'
Source 'https://wooga.artifactoryonline.com/wooga/api/nuget/nuget-private' exception: System.Exception: Error file for Wooga.Tracking.Unity3D.Source exists at /Users/devboy/.local/share/NuGet/Cache/Wooga.Tracking.Unity3D.Source.12.2.0.s1037189022.json.failed
  at Paket.NuGetV2+getDetailsFromNuget@293-2.Invoke (System.String message) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.PrintfImpl+StringPrintfEnv`1[Microsoft.FSharp.Core.Unit].Finalize () [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.PrintfImpl+Final2@235[Microsoft.FSharp.Core.Unit,System.String,Microsoft.FSharp.Core.Unit,System.String,System.String].Invoke (Microsoft.FSharp.Core.FSharpFunc`2 env, System.String a, System.String b) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.OptimizedClosures+Invoke@3041-1[System.String,System.String,Microsoft.FSharp.Core.Unit,Microsoft.FSharp.Core.FSharpFunc`2[Microsoft.FSharp.Core.Unit,Microsoft.FSharp.Core.PrintfImpl+PrintfEnv`3[Microsoft.FSharp.Core.Unit,System.String,Microsoft.FSharp.Core.Unit]]].Invoke (System.String u, System.String v) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.OptimizedClosures+Invoke@3028[System.String,Microsoft.FSharp.Core.Unit,System.String].Invoke (System.String u) [0x00000] in <filename unknown>:0
  at Paket.NuGetV2+getDetailsFromNuget@293-4.Invoke (System.String arg20) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.FSharpFunc`2[System.String,System.String].InvokeFast[Unit] (Microsoft.FSharp.Core.FSharpFunc`2 func, System.String arg1, System.String arg2) [0x00000] in <filename unknown>:0
  at Paket.NuGetV2+getDetailsFromNuget@292-1.Invoke (Microsoft.FSharp.Core.Unit unitVar) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Control.AsyncBuilderImpl+callA@803[Paket.NuGetV2+NugetPackageCache,Microsoft.FSharp.Core.Unit].Invoke (Microsoft.FSharp.Control.AsyncParams`1 args) [0x00000] in <filename unknown>:0
```

**Comment by devboy** [comment]
I think that could be the same issue as this one: https://github.com/fsprojects/Paket/issues/761

**Comment by devboy** [comment]
Fixed in 1.4.7. Thanks a lot!
